server/src/main/java/org/elasticsearch/indices/recovery/plan/SnapshotsRecoveryPlannerService.java

```diff
@@ -126,6 +125,16 @@ private ShardRecoveryPlan computeRecoveryPlanWithSnapshots(@Nullable String shar
         );
     }
 
+    private boolean isSnapshotVersionCompatible(ShardSnapshot snapshot) {
+        Version commitVersion = snapshot.getCommitVersion();
+        // if the snapshotVersion == null that means that the snapshot was taken in a version <= 7.15,
+        // therefore we can safely use that snapshot.
+        if (commitVersion == null) {
+            return Version.CURRENT.luceneVersion.onOrAfter(snapshot.getCommitLuceneVersion());
```

Comment:
Can we add an assert about this too?

```suggestion
            assert Version.V_7_16_0.luceneVersion.onOrAfter(snapshot.getCommitLuceneVersion());
            return Version.CURRENT.luceneVersion.onOrAfter(snapshot.getCommitLuceneVersion());
```